numerics/quadrature.hpp

```diff
@@ -1,32 +1,53 @@
-
+﻿
 #pragma once
 
-#include "quantities/named_quantities.hpp"
-
 #include <type_traits>
 
+#include "geometry/hilbert.hpp"
+#include "quantities/named_quantities.hpp"
+
 namespace principia {
 namespace numerics {
 namespace quadrature {
 namespace internal_quadrature {
 
+using geometry::Hilbert;
 using quantities::Primitive;
 
 template<int points, typename Argument, typename Function>
 Primitive<std::invoke_result_t<Function, Argument>, Argument> GaussLegendre(
-    Function const& function,
+    Function const& f,
+    Argument const& lower_bound,
+    Argument const& upper_bound);
+
+// Computes a Clenshaw-Curtis quadrature on 2ᵖ + 1 points for successive p until
+// the tolerance is satisfied.
+// TODO(egg): The semantics of the |relative_tolerance| are unclear.
+template<int initial_points = 2, typename Argument, typename Function>
+Primitive<std::invoke_result_t<Function, Argument>, Argument>
+AutomaticClenshawCurtis(
+    Function const& f,
+    Argument const& lower_bound,
+    Argument const& upper_bound,
+    double relative_tolerance = 0x1p-16);
+
+// |points| must be of the form 2ᵖ + 1 for some p ∈ ℕ.
```

Comment:
It would be good if the comments explained the difference between this function and the previous one.